### ISSUE TITLE
Query Results get cached even if noCaching is true

### ISSUE THREAD

**Issue by seanhurwitz** [ravendb/ravendb-nodejs-client#467]
![image](https://github.com/user-attachments/assets/f94797a9-7de3-4ccf-9f6e-56ed272dc516)

I have set noCaching to true on each session, but this still gets cached. This is giving us a massive memory leak.
Please help!

**Comment by ml054** [comment]
Hi @seanhurwitz 

Which version of node.js client are you using?

**Comment by seanhurwitz** [comment]
@ml054 6.0.0

I believe the issue is (RequestExecutor.ts):

![image](https://github.com/user-attachments/assets/45784d27-223c-4d0b-8168-60eb57515951)

this._cache is always defined

and the method checks if(cache) cacheResponse(...)

so I suggest: 
responseDispose = await command.processResponse(sessionInfo.noCaching ? undefined : this._cache, response, bodyStream, req.uri as string);

**Comment by seanhurwitz** [comment]
I also set maxHttpCacheSize to 0 but that doesn't help. At least this will on a per session basis. But maybe if the size is 0, it shouldn't initialise the cache at all

**Comment by seanhurwitz** [comment]
node version 20.16.0

**Comment by seanhurwitz** [comment]
![image](https://github.com/user-attachments/assets/e4b279b3-4cf7-461b-9427-f1bdb784d8eb)
I set that cache variable to undefined in my node_modules and it doesn't cache anymore

**Comment by ml054** [comment]
@seanhurwitz 

```
  it("noCaching - on session", async () => {
        const session = store.openSession({
            noCaching: true
        });

        await session.query<BlogPost>({ collection: "blogPosts" })
            .all();

        console.log("noCaching - on session", store.getRequestExecutor().cache.numberOfItems);
        // it prints 1
    });

    it("noCaching - on query", async () => {
        const session = store.openSession();

        await session.query<BlogPost>({ collection: "blogPosts" })
            .noCaching()
            .all();

        console.log("noCaching - on query", store.getRequestExecutor().cache.numberOfItems);
        // it prints 0
    });
```

Please compare those two examples. First uses noCaching as session option, second noCache on query. 

Can you confirm using noCaching on session.query doesn't cause memory leaks? 

I will investiage futher why there is such disprancy and get back to you. I have hope that temporary workaround can unblock your development/deployments for time being.

**Comment by seanhurwitz** [comment]
@ml054  I can confirm it works! Thanks. Please see to the issue though, it's tediuous to say noCaching every time

**Comment by ml054** [comment]
I do agree. Will open a ticket for RavenDB to fix it. Sorry for inconvenience.

**Comment by ml054** [comment]
Reported an issue: https://issues.hibernatingrhinos.com/issue/RavenDB-23188/Non-consistent-noCaching-behavior

**Comment by seanhurwitz** [comment]
thanks. Just to clarify

Store level caching: (ie maxHttpCache = 0) **DOESN'T WORK** 

Session level Caching (ie .openSession({noCaching:true}) **DOESN'T WORK**

***

Query level Caching (.noCaching()) **_DOES WORK_**

Seems the issue was only raised for Sesison Level Caching, can we not also fix it for Store level caching?

**Comment by ml054** [comment]
Sure, reported separate ticket for node.js: https://issues.hibernatingrhinos.com/issue/RDBC-881/Node.js-client-cache-issues

Thanks
